### ISSUE TITLE
Jupyter Notebook: Config Tool Part II

### ISSUE THREAD

**Issue by christinaholtNOAA** [ufs-community/uwtools#644]
# AC

- [ ] Add sections to the config tool notebook covering the compare and validate functions as well as each of the file type classes (e.g. YAMLConfig) and each of their methods.
- [ ] Update `test_config.py` to add unit tests for each new section

# Description

Add to the existing Jupyter notebook for the config tool to document the remaining functions and classes. Unit tests should be added to cover these new sections. 

Let's pause before this work starts and see how we feel about splitting notebooks after the `config realize` notebook is complete. It may end up being a refactoring once the complete body of the work is done. 

# Group discussion

Let's stick with one notebook for now. Revisit later once we've seen the final product (as a separate ticket). Arguments for single notebook:

* Single point of reference in docs
* Single point of reference in future cookbooks
* Without realize, not as exciting.
